controllers/v1/solutions.js

```diff
@@ -16,64 +16,1066 @@ module.exports = class  Solutions extends Abstract{
+   * @returns {Array} Program scope roles.
+   */
 
-        return resolve(createdSolution);
+  async removeEntitiesInScope(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let solutionUpdated = await solutionsHelper.removeEntitiesInScope(
+          req.params._id,
+          req.body.entities,
+          req.userDetails.userToken
+        );
+    
+        return resolve(solutionUpdated);
 
       } catch (error) {
         return reject({
-          status: error.status || httpStatusCode.internal_server_error.status,
-          message: error.message || httpStatusCode.internal_server_error.message,
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
           errorObject: error
         });
       }
     });
   }
 
+
+
+    /**
+  * @api {post} /project/v1/solutions/verifySolution/:Id
+  * @apiVersion 1.0.0
+  * @apiName verify Solutions targeted
+  * @apiGroup Solutions
+  * @apiSampleRequest /project/v1/solutions/verifySolution/5f6853f293734140ccce90cf
+  * @apiHeader {String} X-authenticated-user-token Authenticity token  
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Request:
+  * {
+  *   "role" : "HM,DEO",
+      "entities" : "[236f5cff-c9af-4366-b0b6-253a1789766a"],
+      "entityType" : "1dcbc362-ec4c-4559-9081-e0c2864c2931",
+    }
+  * @apiParamExample {json} Response:
+  * {
+      "message": "Solution verified successfully",
+      "status": 200,
+      "result": {
+          isATargetedSolution : true/false,
+          _id : “5f6853f293734140ccce90cf”,
+      }
+    }
+  */
+
+   /**
+   * verify Solution
+   * @method
+   * @name verifySolution
+   * @param {Object} req - requested data.
+   * @param {String} req.params._id - solution id
+   * @returns {Array}
+   */
+
+   async verifySolution(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        
+        let solutionData = await solutionsHelper.isTargetedBasedOnUserProfile(
+          req.params._id,
+          req.body,
+        );
+
+        return resolve(solutionData);
+
+      }
+      catch (error) {
+        return reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        })
+      }
+    })
+  }
+
+
+  /**
+    * @api {post} /project/v1/solutions/forUserRoleAndLocation?programId=:programId&type=:type&subType=:subType&page=:page&limit=:limit Auto targeted solutions
+    * @apiVersion 1.0.0
+    * @apiName Auto targeted solution
+    * @apiGroup Solutions
+    * @apiParamExample {json} Request-Body:
+    * {
+        "role" : "HM,DEO",
+   		  "entities" : "[236f5cff-c9af-4366-b0b6-253a1789766a]",
+        "entityType" : "1dcbc362-ec4c-4559-9081-e0c2864c2931",
```

Comment:
@Prajwal17Tunerlabs please correct the value of entitytype

Reply:
changed the value of entityType